src/EnergyPlus/WindowManager.cc

```diff
@@ -6864,88 +7093,134 @@ namespace WindowManager {
-                                           pow_3(state.dataWindowManager->thetas(6) + state.dataWindowManager->thetas(7));
-                Rbare = 1.0 / state.dataWindowManager->scon(1) + 1.0 / hGapTot(1) + 1.0 / state.dataWindowManager->scon(2) + 1.0 / hGapTot(2) +
-                        1.0 / state.dataWindowManager->scon(3) + 1.0 / hGapTot(3) + 1.0 / state.dataWindowManager->scon(4);
-                state.dataWindowManager->Rtot = rOut + Rbare + rIn;
-                SHGC = AbsBeamNorm(1) * (rOut + 0.5 / state.dataWindowManager->scon(1)) / state.dataWindowManager->Rtot +
-                       AbsBeamNorm(2) * (rOut + 1.0 / state.dataWindowManager->scon(1) + 1.0 / hGapTot(1) + 0.5 / state.dataWindowManager->scon(2)) /
-                           state.dataWindowManager->Rtot +
-                       AbsBeamNorm(3) *
-                           (rOut + 1.0 / state.dataWindowManager->scon(1) + 1.0 / hGapTot(1) + 1.0 / state.dataWindowManager->scon(2) +
-                            1.0 / hGapTot(2) + 0.5 / state.dataWindowManager->scon(3)) /
-                           state.dataWindowManager->Rtot +
-                       AbsBeamNorm(4) *
-                           (rOut + 1.0 / state.dataWindowManager->scon(1) + 1.0 / hGapTot(1) + 1.0 / state.dataWindowManager->scon(2) +
-                            1.0 / hGapTot(2) + 1.0 / state.dataWindowManager->scon(3) + 1.0 / hGapTot(3) + 0.5 / state.dataWindowManager->scon(4)) /
-                           state.dataWindowManager->Rtot; // CR7682
-                SHGC += AbsBeamShadeNorm;
-                SHGC += TSolNorm;
+        // Get center-of-glass conductance and solar heat gain coefficient
+        // including inside and outside air films
+        Real64 inputU = state.dataMaterial->Material(state.dataConstruction->Construct(ConstrNum).LayerPoint(1)).SimpleWindowUfactor;
+
+        // Calculate the NominalConductance glazing only
+        EvalNominalWindowCond(state, AbsBeamShadeNorm, AbsBeamNorm, hgap, NominalConductance, SHGC, TSolNorm);
+
+        if (WinterSummerFlag == 1) {
+            state.dataHeatBal->NominalUGlazingOnly = NominalConductance;
+            if (inputU > 0) { // only compute adjustment ratio when there is valid user input U
+                Real64 CoeffAdjRatio = 1;
+                Real64 hcinRated = state.dataWindowManager->hcin;
+                // Adjustment ratio applies to convective film coefficients when input U value is above the limit of the simple glazing nominal U
+                // Representing the nominal highly conductive frame effects. Solved iteratively.
+                int MaxIter = 10;
+                while (inputU - NominalConductance > 0.01 && MaxIter > 0) {
+                    state.dataWindowManager->hcout *= CoeffAdjRatio;
+                    state.dataWindowManager->hcin *= CoeffAdjRatio;
+                    EvalNominalWindowCond(state, AbsBeamShadeNorm, AbsBeamNorm, hgap, NominalConductance, SHGC, TSolNorm);
```

Comment:
you have to update the window layer temperatures inside this loop.   Look at the functional dependency on thetas inside EvalNominalWindowCond....  need to call WindowTempsForNominalCond to get the new layer temperatures.

Reply:
The while loop is essentially solving the following equation in a recursive fashion. 
![image](https://user-images.githubusercontent.com/13421161/130456633-a4688a47-e65b-4aba-ba30-679c7da27989.png)
I believe the thetas(i) should be constant during the solving process.